lib/Transforms/UnifyKernels.cpp

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright HeteroCL authors. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <iostream>
+
+#include "PassDetail.h"
+
+#include "hcl/Dialect/HeteroCLDialect.h"
+#include "hcl/Dialect/HeteroCLOps.h"
+#include "hcl/Dialect/HeteroCLTypes.h"
+#include "hcl/Dialect/TransformOps/HCLTransformOps.h"
+#include "hcl/Transforms/Passes.h"
+
+#include "hcl-c/Dialect/Dialects.h"
+#include "mlir/CAPI/IR.h"
+
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/IR/AffineMap.h"
+#include "mlir/IR/IRMapping.h"
+#include "mlir/IR/OperationSupport.h"
+#include "mlir/InitAllDialects.h"
+
+using namespace mlir;
+using namespace hcl;
+
+namespace mlir {
+namespace hcl {
+
+bool compareAffineExprs(AffineExpr lhsExpr, AffineExpr rhsExpr) {
```

Comment:
One question, did you do canonicalization somewhere? If I pass in `d0+1` and `1+d0`, will this function return true?

Reply:
Yeah, I think that would be an issue for now. As I commented "Todo" in compareAffineMaps function, this is now a temporary solution. I was thinking about use a evaluation strategy to match some more sophisticated case as you mentioned. I will test and see how it work in future pr.

Reply:
> One question, did you do canonicalization somewhere? If I pass in `d0+1` and `1+d0`, will this function return true?

Yeah, I think that would be an issue for now. As I commented "Todo" in compareAffineMaps function, this is now a temporary solution. I was thinking about use a evaluation strategy to match some more sophisticated case as you mentioned. I will test and see how it work in future pr.